Lib/test/test_readline.py

```diff
@@ -9,14 +9,21 @@
 import sys
 import tempfile
 import unittest
-from test.support import import_module, unlink, temp_dir, TESTFN
+from test.support import import_module, unlink, temp_dir, TESTFN, verbose
 from test.support.script_helper import assert_python_ok
 
 # Skip tests if there is no readline module
 readline = import_module('readline')
 
 is_editline = readline.__doc__ and "libedit" in readline.__doc__
 
+
+def setUpModule():
+    if verbose and hasattr(readline, "_READLINE_VERSION"):
+        print(f"readline version: {readline._READLINE_VERSION:#x}")
+        print(f"readline runtime version: {readline._READLINE_RUNTIME_VERSION:#x}")
```

Comment:
> So maybe we don't need the hasattr check?

I like the idea of using test_readline unchanged on Python implementations other than CPython. Tests cannot rely on *private* attributes in tests, not without @cpython_only or hasattr(). I added a comment to explain the rationale.

> is_editline

Right, I also logged this flag, even if the other attributes are not available.

I chose to also expose rl_library_version as a private attribute to be able to log it in test_readline. It may help to debug subtle differences between two readline versions, and validate manually is_editline flag.